pkg/apis/caching/v1alpha1/image_types_test.go

```diff
@@ -0,0 +1,148 @@
+	cases := []struct {
+		name    string
+		status  ImageStatus
+		isReady bool
+	}{{
+		name:    "empty status should not be ready",
+		status:  ImageStatus{},
+		isReady: false,
+	}, {
+		name: "Different condition type should not be ready",
+		status: ImageStatus{
+			Conditions: []ImageCondition{{
+				Type:   "foo",
+				Status: corev1.ConditionTrue,
+			}},
+		},
+		isReady: false,
+	}, {
+		name: "False condition status should not be ready",
+		status: ImageStatus{
+			Conditions: []ImageCondition{{
+				Type:   ImageConditionReady,
+				Status: corev1.ConditionFalse,
+			}},
+		},
+		isReady: false,
+	}, {
+		name: "Unknown condition status should not be ready",
+		status: ImageStatus{
+			Conditions: []ImageCondition{{
+				Type:   ImageConditionReady,
+				Status: corev1.ConditionUnknown,
+			}},
+		},
+		isReady: false,
+	}, {
+		name: "Missing condition status should not be ready",
+		status: ImageStatus{
+			Conditions: []ImageCondition{{
+				Type: ImageConditionReady,
+			}},
+		},
+		isReady: false,
+	}, {
+		name: "True condition status should be ready",
+		status: ImageStatus{
+			Conditions: []ImageCondition{{
+				Type:   ImageConditionReady,
+				Status: corev1.ConditionTrue,
+			}},
+		},
+		isReady: true,
+	}, {
+		name: "Multiple conditions with ready status should be ready",
+		status: ImageStatus{
+			Conditions: []ImageCondition{{
+				Type:   "foo",
+				Status: corev1.ConditionTrue,
+			}, {
+				Type:   ImageConditionReady,
+				Status: corev1.ConditionTrue,
+			}},
+		},
+		isReady: true,
+	}, {
+		name: "Multiple conditions with ready status false should not be ready",
+		status: ImageStatus{
+			Conditions: []ImageCondition{{
+				Type:   "foo",
+				Status: corev1.ConditionTrue,
+			}, {
+				Type:   ImageConditionReady,
+				Status: corev1.ConditionFalse,
+			}},
+		},
+		isReady: false,
+	}}
+
+	for _, tc := range cases {
+		if e, a := tc.isReady, tc.status.IsReady(); e != a {
+			t.Errorf("%q expected: %v got: %v", tc.name, e, a)
+		}
+	}
+}
+
+func TestImageConditions(t *testing.T) {
+	rev := &Image{}
+	foo := &ImageCondition{
+		Type:   "Foo",
+		Status: "True",
+	}
+	bar := &ImageCondition{
+		Type:   "Bar",
+		Status: "True",
+	}
+
+	// Add a new condition.
+	rev.Status.SetCondition(foo)
+
+	if got, want := len(rev.Status.Conditions), 1; got != want {
+		t.Fatalf("Unexpected Condition length; got %d, want %d", got, want)
+	}
+
+	// Add nothing
+	rev.Status.SetCondition(nil)
+
+	if got, want := len(rev.Status.Conditions), 1; got != want {
+		t.Fatalf("Unexpected Condition length; got %d, want %d", got, want)
+	}
+
+	// Add a second condition.
+	rev.Status.SetCondition(bar)
+
+	if got, want := len(rev.Status.Conditions), 2; got != want {
+		t.Fatalf("Unexpected Condition length; got %d, want %d", got, want)
+	}
+
+	// Add nil condition.
+	rev.Status.SetCondition(nil)
+
+	if got, want := len(rev.Status.Conditions), 2; got != want {
+		t.Fatalf("Unexpected Condition length; got %d, want %d", got, want)
+	}
```

Comment:
You don't have a test for the `VolatileTime` behaviour

Reply:
Done